articles/understanding-dart-loops/index.md

```diff
@@ -0,0 +1,340 @@
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+
+Here the value of the counter variable will increase by one from 0 based on the increment specified. If the value of i becomes equal to four or more, it ends since the condition states that the value of i should be less than 4. t prints hello when it meets the condition, hence the output is hello written four times (0,1,2,3) < 4. 
+
+#### While Loop
+
+**Syntax of While Loop.**
+```Dart
+void Main(){
+   // initialize counter variable;
+   
+   while(condition){
+   
+      //code;
+      
+      increment/decrement;
+    }
+}
+```
+
+**How the While Loop works:**
+
+**Condition Check - Code Execute - Increment**.
+
+1. We initialization of the counter variable outside the syntax. 
+
+2. It begins with the condition check, if it's true, the code executes. 
+
+3. The value of the counter variable then increments or decrements, and the new counter variable goes through the condition check. 
+
+4. It will continue until the condition check is false.
+
+> NOTE: 
+> We must always have the increment or decrement part of the While Loop, failure to which will create a never-ending loop.
+
+***Example:***
+
+```Dart
+void main(){
+  int i = 0;
+   while(i < 3){
+      print("Hello");
+      i++;
+   }
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+```
+
+Our condition check begins with 0<3 which is true therefore the code executes and prints out the first hello. The value of i increases by one to 1 and the condition 1<3 is still true, so the code executes. It goes, the new variable becomes (2<3) and ends when the condition is now (3<3) which is false. Hence three hellos get printed. 
+
+#### Do-While Loop
+
+The do-While loop is an example of [Indefinite Loops](#indefinite-loops).
+
+**Syntax of Do-While Loop:**
+
+```Dart
+void main(){
+   // initialize counter variable
+
+   do {
+      //code;
+     
+      increment/decrement;
+   } while(condition);
+}
+```
+
+**How the Do-While Loop works:**
+
+**Code Execution - Increment - Condition Check**.
+
+1. Code executes then the counter variable initialized outside the syntax increments or decrements. 
+
+2. The condition check - if it's met, it begins from code execution. 
+
+3. The code execution will continue until the condition check turns false. 
+
+> Note that the code block in the Do-While loop executes at least once.
+
+***Example:***
+
+```Dart
+void main(){
+   int i = 0;
+   do {
+      print ("Hello");
+      i++;
+   } while (i<5);
+}
+```
+
+Output:
+
+```
+Hello
+Hello
+Hello
+Hello
+Hello
+```
+
+The code first executes then the counter variable increments by one. It then checks the condition (1<5). It is true, so the loop will begin again. The code executes, then the counter variable increments by one. It then checks the condition (2<5). It meets the condition. It will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.
```

Comment:
It will continue until the code executes and the new counter variable is 5 and the condition checked is (5<5) which is false, so the loop ends after printing 5 hellos.

Rewrite this line as:

This will continue until the value of the counter variable is 5. This time the condition check (5<5) will be evaluated as false, so the loop ends after printing 5 hellos.